mlapi/discriminating_algo.py

```diff
@@ -0,0 +1,129 @@
+import statistics
+
+class DiscriminatingAlgo(object):
+    # Discrimination parameters
+    facets_sample_count = 3  # minimum docs by facet to be accepted
+    tolerance_interval = 10 # max standard deviation accepted ()
+    nbr_facet_values = 2 # to consider a facet as having a large number of values
+
+    facets = [] # names
+    step2 = {}
+    step3 = {}
+
+    def discriminating_facets_algo(self, facets_by_document):
+        documents_by_facet = self.inverse_dictionary(facets_by_document)
+        data = self.get_formated_data(documents_by_facet)
+        self.execute_algorithm(data)
+        if not (len(self.step3)):
+            self.adjust_parameters()
+            self.execute_algorithm(data)
+        return self.get_values_by_facet(self.step3)
+
+    def inverse_dictionary(self, facets_by_document):
```

Comment:
Pythonic way to do this: https://stackoverflow.com/questions/483666/python-reverse-invert-a-mapping

You can probably replace the whole function with this solution

Reply:
I can't use : documents_by_facet = {v: k for k, v in facets_by_document.items()}
because the dictionary elements are not unique